From ce2963d82620bc47e26dd61335428a325b64de27 Mon Sep 17 00:00:00 2001
From: Marco Angerer <marco.angerer@fau.de>
Date: Mon, 16 Dec 2024 20:51:04 +0100
Subject: [PATCH] Ensure installabillty on ILIAS 8

---
 classes/class.assSQLQuestion.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/classes/class.assSQLQuestion.php b/classes/class.assSQLQuestion.php
index 8165e35..2738757 100644
--- a/classes/class.assSQLQuestion.php
+++ b/classes/class.assSQLQuestion.php
@@ -252,7 +252,7 @@ class assSQLQuestion extends assQuestion
         $this->setComment((string) $data['description']);
         $this->setSuggestedSolution((string) $data['solution_hint']);
 
-        $this->setQuestion(ilRTE::_replaceMediaObjectImageSrc($data['question_text'], 1));
+        $this->setQuestion((string) ilRTE::_replaceMediaObjectImageSrc($data['question_text'], 1));
     }
 
     /**
@@ -525,7 +525,7 @@ class assSQLQuestion extends assQuestion
      *
      * @throws ilTestException
      */
-    public function calculateReachedPoints($active_id, $pass = null, $authorizedSolution = true, $returndetails = false): array|float
+    public function calculateReachedPoints($active_id, $pass = null, $authorizedSolution = true, $returndetails = false)
     {
         if ($returndetails) {
             throw new ilTestException('return details not implemented for '.__METHOD__);
-- 
2.39.5

